app/src/main/java/com/example/nicocalculator/MainActivity.kt

```diff
@@ -2,33 +2,62 @@ package com.example.nicocalculator
 
 import android.os.Bundle
 import android.support.v7.app.AppCompatActivity
+import android.widget.Button
+import com.example.nicocalculator.mvp.model.CalculatorModel
+import com.example.nicocalculator.mvp.presenter.Presenter
+import com.example.nicocalculator.mvp.view.KeyView
 import kotlinx.android.synthetic.main.activity_main.*
 
 class MainActivity : AppCompatActivity() {
 
 
-
     companion object {
         const val OPERATIONS_DEFAULT_TEXT = "Operations"
         const val RESULTS_DEFAULT_TEXT = "Results"
-        const val SET_VALUE_PRESSED = "Pressed"
-        const val SET_VALUE_EMPTY = "Empty"
     }
 
+    lateinit var presenter: Presenter
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
-        text_results.text = RESULTS_DEFAULT_TEXT
-        text_operations.text = OPERATIONS_DEFAULT_TEXT
+        presenter = Presenter(CalculatorModel(RESULTS_DEFAULT_TEXT, OPERATIONS_DEFAULT_TEXT), KeyView(this))
 
-        button_0.setOnClickListener {
-            text_results.text = SET_VALUE_PRESSED
+        button_delete.setOnClickListener {
+            presenter.onDeleteButtonPressed()
+        }
+        button_erase.setOnClickListener {
+            presenter.onEraseButtonPressed()
         }
 
-        button_delete.setOnClickListener {
-            text_results.text = SET_VALUE_EMPTY
+        button_equal.setOnClickListener {
+            presenter.onEqualButtonPressed()
         }
 
+        val buttons = mutableListOf<Button>(
+            button_0, button_1, button_2, button_3, button_4,
+            button_5, button_6, button_7, button_8, button_9,
+            button_close_par, button_open_par, button_minus, button_plus,
+            button_mult, button_div, button_dot
+        )
+
+        for (button in buttons) {
+            button.setOnClickListener {
+                presenter.onKeyButtonPressed(button)
+            }
+        }
     }
 }
+
+/*
```

Comment:
If you will not use this code please remove it

Reply:
Done